src/main/java/com/google/api/codegen/config/InterfaceConfig.java

```diff
@@ -14,381 +14,10 @@
-    }
-    return methodConfig;
-  }
-
-  public boolean hasDefaultServiceAddress() {
-    return !getRequiredConstructorParams().contains(SERVICE_ADDRESS_PARAM);
-  }
-
-  public boolean hasDefaultServiceScopes() {
-    return !getRequiredConstructorParams().contains(SCOPES_PARAM);
-  }
-
-  public boolean hasDefaultInstance() {
-    return getRequiredConstructorParams().size() == 0;
-  }
-
-  /**
-   * If rerouteToGrpcInterface is set, then looks up that interface and returns it, otherwise
-   * returns the value of defaultInterface.
-   */
-  public static Interface getTargetInterface(
-      Interface defaultInterface, String rerouteToGrpcInterface) {
-    Interface targetInterface = defaultInterface;
-    if (!Strings.isNullOrEmpty(rerouteToGrpcInterface)) {
-      targetInterface =
-          defaultInterface.getModel().getSymbolTable().lookupInterface(rerouteToGrpcInterface);
-      if (targetInterface == null) {
-        throw new IllegalArgumentException(
-            "reroute_to_grpc_interface not found: " + rerouteToGrpcInterface);
-      }
-    }
-    return targetInterface;
-  }
-
-  /** Creates a list of fields that can be turned into IAM resources */
-  private static ImmutableList<Field> createIamResources(
-      Model model, List<IamResourceProto> resources) {
-    ImmutableList.Builder<Field> fields = ImmutableList.builder();
-    for (IamResourceProto resource : resources) {
-      TypeRef type = model.getSymbolTable().lookupType(resource.getType());
-      if (type == null) {
-        throw new IllegalArgumentException("type not found: " + resource.getType());
-      }
-      if (!type.isMessage()) {
-        throw new IllegalArgumentException("type must be a message: " + type);
-      }
-      Field field = type.getMessageType().lookupField(resource.getField());
-      if (field == null) {
-        throw new IllegalArgumentException(
-            String.format(
-                "type %s does not have field %s", resource.getType(), resource.getField()));
-      }
-      fields.add(field);
-    }
-    return fields.build();
-  }
-
-  public boolean hasPageStreamingMethods() {
-    for (MethodConfig methodConfig : getMethodConfigs()) {
-      if (methodConfig.isPageStreaming()) {
-        return true;
-      }
-    }
-    return false;
-  }
-
-  public boolean hasBatchingMethods() {
-    for (MethodConfig methodConfig : getMethodConfigs()) {
-      if (methodConfig.isBatching()) {
-        return true;
-      }
-    }
-    return false;
-  }
-
-  public boolean hasGrpcStreamingMethods() {
-    for (MethodConfig methodConfig : getMethodConfigs()) {
-      if (methodConfig.isGrpcStreaming()) {
-        return true;
-      }
-    }
-    return false;
-  }
-
-  public boolean hasLongRunningOperations() {
-    for (MethodConfig methodConfig : getMethodConfigs()) {
-      if (methodConfig.isLongRunningOperation()) {
-        return true;
-      }
-    }
-    return false;
-  }
+public interface InterfaceConfig {
```

Comment:
`SurfaceNamer` definitely started as GAPIC-based, but over time, I want to convert it to support other inputs. Basically I don't want to repeat all of the naming work for each input. 

Background of this PR: I was starting to bootstrap Discovery-based GAPIC generation, but then encountered the terminology hell. I started fixing the terminology, but some of the generalization work came along for the ride. This interface needs to exist because I have a local `DiscoInterfaceConfig` that is part of the bootstrapping.